### PR TITLE
Add metadata + dependency upgrades

### PR DESCRIPTION
**Pull request by edorivai**
# tl;dr - Adds metadata config

Usage example:

```js
const googleMulter = new MulterGoogleCloudStorage({
  metadata: {
    cacheControl: 'public, max-age=31536000'
  },
});
```

# Context

My original goal was to be able allow for the cache-control metadata field to be set. While I was trying to implement it, I ran into a bunch of issues which led me to cascade a bunch of package version upgrades.

Most notably; I've bumped `@google-cloud/storage` from 1.1.1 to 2.5.0, this also allowed me to remove `@types/google-cloud__storage`, since TS definition are now shipped within the official package :tada: 

I've also bumped the version of TypeScript to 3.5.2, since the 2.x version had some problems with the newer version of `@google-cloud/storage`.

Finally, there are some TS errors that I fixed, like adding `acl` to the constructor options.

### CODE REVIEW

**Comment by edorivai** [comment]
https://github.com/ARozar/multer-google-storage/pull/13/commits/79eaa7a6f8fb64e237e03baa4bf9abbc09fd041b updates the Node versions in the Travis tests to exclude EOL versions, and include the `current` and LTS versions.
